modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -916,7 +931,7 @@ def join(self, other, how="inner", on=None, sort=False, suffixes=("_x", "_y")):
 
         if sort:
             res = res.sort_rows(
-                on, ascending=True, ignore_index=True, na_position="last"
+                left_on, ascending=True, ignore_index=True, na_position="last"
```

Comment:
This doesn't seem correct in the case `how` is equal to `right` or `outer`.

Reply:
OmniSci engine doesn't support both of them

Reply:
If it is so, shouldn't we default to pandas in the cases? I don't see we handle it for OmniSci engine.

Reply:
Yes, we should. Currently, we would just get a runtime error from OmniSci.

Reply:
Ok, let's default to pandas as part of separate PR.

Reply:
Another question. How will it work if `left_on` and `right_on` are multindex levels but not columns?

Reply:
We don't  cover multiindex levels yet in OmniSci backend